src/Tree/BruteForceTree.cpp

```diff
@@ -461,25 +462,33 @@ void BruteForceTree<ndim,ParticleType,TreeCell>::StockTreeProperties
 //=================================================================================================
 template <int ndim, template<int> class ParticleType, template<int> class TreeCell>
 void BruteForceTree<ndim,ParticleType,TreeCell>::UpdateActiveParticleCounters
- (ParticleType<ndim> *partdata)        ///< [in] Main particle array
+ (Particle<ndim> *part_gen)            ///< [in] Main particle array
 {
   int i;                               // SPH particle index
+  int c;                               // Cell index
   int ilast;                           // Last particle in linked list
 
   debug2("[BruteForceTree::UpdateActiveParticleCounters]");
 
+  ParticleType<ndim>* partdata = reinterpret_cast<ParticleType<ndim>*>(part_gen);
 
-  celldata[0].Nactive = 0;
-  i = celldata[0].ifirst;
-  ilast = celldata[0].ilast;
+#pragma omp parallel for default(none) private(c,i,ilast) shared(partdata)
+  for (c=0; c<Ncell; c++) {
+    celldata[c].Nactive = 0;
 
-  // Else walk through linked list to obtain list and number of active ptcls.
-  while (i != -1) {
-    if (i < Ntot && partdata[i].flags.check_flag(active) && !partdata[i].flags.is_dead())
-      celldata[0].Nactive++;
-    if (i == ilast) break;
-    i = inext[i];
-  };
+    if (celldata[c].level != ltot) continue;
+    i = celldata[c].ifirst;
+    ilast = celldata[c].ilast;
+
+    // Else walk through linked list to obtain list and number of active ptcls.
+    while (i != -1) {
+      if (i < Ntot && partdata[i].flags.check_flag(active) && !partdata[i].flags.is_dead())
+        celldata[c].Nactive++;
+      if (i == ilast) break;
+      i = inext[i];
+    };
+
+  }
```

Comment:
We only need the active particle list for the leaf cells since those are what is returned by compute active cell list, so I believe this now correct. I don't know why it was working before.